api/v1/cluster_types.go

```diff
@@ -626,6 +630,23 @@ type ManagedRoles struct {
 	PasswordStatus map[string]PasswordState `json:"passwordStatus,omitempty"`
 }
 
+// TablespaceStatus represents the status of a tablespace in the cluster
+type TablespaceStatus string
+
+const (
+	// TablespaceStatusReconciled indicates the tablespace in DB matches the Spec
+	TablespaceStatusReconciled TablespaceStatus = "reconciled"
+	// TablespaceStatusPendingReconciliation indicates the tablespace in Spec requires creation in the DB
+	TablespaceStatusPendingReconciliation TablespaceStatus = "pending"
```

Comment:
So, at the moment an error would manifest as the TBSP in "pending" and then error logs.
On having a richer "error signaling" as we do in managed roles.
I thought about that. But I don't know what bits we could have where a reasonable request could not be honored by Postgres.
If we have the filesystem mounted and the PVC ready, is there any case where the tablespace creation would fail?

Reply:
I was having a look in the [postgres codebase](https://github.com/postgres/postgres/blob/afc3b0143c89405540461f306137ab6a9671c3bb/src/backend/commands/tablespace.c#L214), and it looks like we should have covered every case. But things can change... it's better to be prepared for it.

What's your opinion?

Reply:
It sounds like there should not be many possibilities for a `CREATE TABLESPACE` to go wrong once the PVC is correctly mounted. So for now, let's avoid adding more complexity to this PR.